### DISCUSSION TITLE
might be of help for the button issue

### DISCUSSION THREAD

**Discussion by Vertigo1207** [NoMeansNowastaken/KarutaSniper#14] [Ideas]
why would the button interaction be broken
Without more context or error messages, it's hard to say for certain what could be causing the issue. However, here are a few common reasons why a button interaction might not be working:

Incorrect ID: Ensure that the ID of the button you are trying to interact with is correct. Double-check that the ID matches in both the code and the Discord client.

Missing or Incorrect Permissions: Ensure that the bot has the necessary permissions to interact with the button. For example, the bot needs the MANAGE_MESSAGES permission to delete messages, which is necessary for removing a button once it has been clicked.

Not registering the interaction: Ensure that the button interaction is properly registered by the bot. You can use the wait_for method to listen for interaction events and respond accordingly.

Incorrect Event Name: Ensure that the event name matches the interaction event you are trying to listen for. For example, if you are trying to listen for a button click, the event name should be on_button_click.

API version mismatch: Ensure that the API version of the bot matches the version of the API that the Discord client is using. If they are different, the interaction may not work properly.

If you are still having trouble, try checking the Discord API documentation or the documentation for the library you are using to see if there are any known issues or troubleshooting steps.

**Comment by NoMeansNowastaken** [comment]
i rewrote button code in the beta branch, however i fucked the lambdas up so they error and that shit straight outa chatgpt
error i need to fix below:
```Ignoring exception in on_message
Traceback (most recent call last):
  File "C:\Users\user\AppData\Roaming\Python\Python310\site-packages\discord\client.py", line 417, in _run_event
    await coro(*args, **kwargs)
  File "C:\Users\user\PycharmProjects\karuta bot\main.py", line 262, in on_message
    reaction = await self.wait_for('reaction_add', check=check)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\lib\asyncio\tasks.py", line 408, in wait_for
    return await fut
  File "C:\Users\user\AppData\Roaming\Python\Python310\site-packages\discord\client.py", line 450, in dispatch
    result = condition(*args)
TypeError: Main.on_message.<locals>.check() takes 1 positional argument but 2 were given

**Comment by Vertigo1207** [comment]
bahaahhahaha
